### PR TITLE
SI-2866-check-contract-w-privs

### PR DESCRIPTION
**Pull request by Allyson-English**

### CODE REVIEW

File: internal/auth/auth.go [hunk 1]
```diff
@@ -54,3 +41,47 @@ func RequiresTokenCheck(ctx context.Context, _ any, next graphql.Resolver) (any,
...
+	for _, priv := range privs {
+		if _, ok := claim.privileges[priv]; !ok {
+			return nil, fmt.Errorf("%w: missing required privilege %s", errUnauthorized, priv)
+		}
+	}
+
+	return next(ctx)
+}
+
+// func (pcv *PrivilegeContractValidator) ManufacturerNFTPrivCheck(ctx context.Context, _ any, next graphql.Resolver, privs []model.Privilege) (any, error) {
```
Comment (KevinJoiner) [inline internal/auth/auth.go:70]: When we implement this I would like to combine the duplicate logic in its own function.

**Review by KevinJoiner — APPROVED** [review]

**Comment by KevinJoiner** [comment]
run `make tools` then `make gql`. I think you are generating with older binaries, which is why CI is failing.

**Comment by elffjs** [comment]
> run `make tools` then `make gql`. I think you are generating with older binaries, which is why CI is failing.

We have learned that this doesn't work well with the macOS version of Make, which is sort of frozen in time. But!

```
brew install make
gmake tools
gmake gql
```

**Comment by elffjs** [comment]
I'm still weirded out by how large the diff is.

One sec.

File: internal/auth/auth.go (hunk 1 above)
Comment (elffjs) [inline internal/auth/auth.go:70]: Does this have any reason to be in the same struct as the other one?

File: internal/auth/auth.go [hunk 2]
```diff
@@ -54,3 +41,47 @@ func RequiresTokenCheck(ctx context.Context, _ any, next graphql.Resolver) (any,
 	}
 	return next(ctx)
 }
+
+type PrivilegeContractValidator struct {
+	VehicleNFTAddress common.Address
+	// ManufAddr         string
+}
+
+// VehicleNFTPrivCheck checks if the claim set in the context includes the correct address the required privileges for the VehicleNFT contract.
```
Comment (elffjs) [inline internal/auth/auth.go:50]: I think this sentence could be better.

**Review by elffjs — APPROVED** [review]

**Comment by elffjs** [comment]
Normally I would complain about all the `//` but "fast-follow", as they say.

**Pull request merged**